src/axom/quest/ArrayIndexer.hpp

```diff
@@ -26,23 +37,92 @@ class ArrayIndexer
 
 public:
   /*!
-    @brief Constructor for row- or column major indexing.
-    @param [in] lengths Lengths of the array
+    @brief Constructor for row- or column-major indexing.
+    @param [in] shape Shape of the array
+    @param [in] order: c is column major; r is row major.
```

Comment:
Does this arg correspond to the values in the `ArrayStrideOrder` struct above? If so, I think this should be completely clear from the comments and variable name (e.g., order -> array_stride_order). Also, would some error checking on the value given by the caller help?